assets/js/modules/analytics-4/datastore/settings.js

```diff
@@ -87,7 +87,7 @@ export async function submitChanges( { select, dispatch } ) {
 		dispatch( MODULES_ANALYTICS_4 ).setWebDataStreamID( webdatastream._id );
 		dispatch( MODULES_ANALYTICS_4 ).setMeasurementID(
 			// eslint-disable-next-line sitekit/acronym-case
-			webdatastream.measurementId
+			webdatastream.webStreamData.measurementId
```

Comment:
Maybe safer to use `?.` here? Not strictly necessary since we only deal with web data streams, but you _do_ use it below, so not sure why we shouldn't have it here.
```suggestion
			webdatastream.webStreamData?.measurementId
```

Reply:
`webdatastream` here comes from our `createWebDataStream` action so we should be able to trust that this is web dataStream object as long as there was no error as we do here. Adding the conditional chain here would move the error raised to `setMeasurementID` because `invariant` within will not allow an `undefined` value to be passed. So I think it might be more useful to let this error here if `webStreamData` was missing for some reason although it never should be if `createWebDataStream` was successful. We could add a check in that action perhaps?

The other instance where a conditional chain was used already had one, so I was following what was already there. I'll leave a separate comment on that one.